### DISCUSSION TITLE
Logger level precedence discrepancy

### DISCUSSION THREAD

**Discussion by somethingSTRANGE** [verdaccio/verdaccio#4654] [Q&A] [answered]
The [logger](https://verdaccio.org/docs/logger/) documentation provides the list of options for the `level` property as `[fatal, error, warn, http, info, debug, trace]`

My assumption is that the option order was meant to show precedence. Anything to the left of the selected option will be included along with the option itself.

* `warn` will include all `fatal`, `error`, and `warn` messages
* `http` will include all `fatal`, `error`, `warn`, and `http` messages
* `info` will include all `fatal`, `error`, `warn`, `http`, and `info` messages
* `trace` will include all message types

In actuality, setting `level: info` will include only `fatal`, `error`, `warn`, and `info`  with `http` being excluded.

Looking at the code, it looks like the precedence is different.

* `fatal = 60`, `error = 50`, `warn = 40`, `info = 30`, `http = 25`, `debug = 20`, `trace = 10`

If the documentation was intending to show option precedence, then either this is a bug in code or the documentation should be updated with the correct level option order.

**Comment by mbtools** [comment, marked as answer]
You are correct about the precedence: `http` is between `info` and `debug`. Feel free to submit a PR to improve the docs 

https://github.com/verdaccio/verdaccio/blob/master/website/docs/logger.md
https://github.com/verdaccio/verdaccio/blob/master/website/versioned_docs/version-5.x/logger.md
https://github.com/verdaccio/verdaccio/blob/master/website/versioned_docs/version-6.x/logger.md

**Comment by somethingSTRANGE** [comment]
Created PR #4723
